crates/sui-graphql-client/src/lib.rs

```diff
@@ -478,6 +502,159 @@ impl Client {
             .map(|c| c.sequence_number))
     }
 
+    // ===========================================================================
+    // Dynamic Field(s) API
+    // ===========================================================================
+
+    /// Access a dynamic field on an object using its name. Names are arbitrary Move values whose
+    /// type have copy, drop, and store, and are specified using their type, and their BCS
+    /// contents, Base64 encoded.
+    ///
+    /// Same as [`dynamic_field`] but instead of passing the bcs bytes, it takes a value of a
+    /// type that implements Serialize. The `type_` parameter is still required and represents the
+    /// dynamic field's type.
+    ///
+    /// This returns [`DynamicFieldOutput`] which contains the name, the value as json, and object.
+    pub async fn dynamic_field_with_name(
```

Comment:
`with_name` is a bit of a confusing name for this, because both functions accept a name, but one accepts it as a serializable type, and the other accepts BCS bytes. My assumption is that people will tend to favour this form, so maybe we name this one `dynamic_field`, and we come up with a name for the BCS version that indicates that it is more primitive? 

... `dynamic_field_with_encoded_name`, `dynamic_field_with_bcs_name?` ? (Naming is hard)

The other option is to adopt the pattern that we suggested for transaction inputs: Introduce a type for dynamic field names, and `Into` impls for that:

```rust

enum Name(Vec<u8>);
pub struct BcsName(Vec<u8>);

impl GraphQlClient {
    pub async fn dynamic_field(
        &self, 
        address: Address, 
        type_: &str, 
        name: impl Into<Name>,
    ) { /* ... */ }
}

impl<T: Serialize> From<T> for Name { /* ... */ }
impl From<BcsName> for Name { /* ... */ };
```

Now you can call it as `client.dynamic_field(addr, "...", my_type)` or `client.dynamic_field(addr, "...", BcsName(...))`, meaning you no longer need to double the API surface area.

Reply:
Added this!